azurerm/internal/services/network/data_source_virtual_machine_scale_set_network_interfaces.go

```diff
@@ -0,0 +1,275 @@
+										Computed: true,
+									},
+
+									"application_gateway_backend_address_pools_ids": {
+										Type:     schema.TypeSet,
+										Computed: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+										Set:      schema.HashString,
+									},
+
+									"load_balancer_backend_address_pools_ids": {
+										Type:     schema.TypeSet,
+										Computed: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+										Set:      schema.HashString,
+									},
+
+									"load_balancer_inbound_nat_rules_ids": {
+										Type:     schema.TypeSet,
+										Computed: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+										Set:      schema.HashString,
+									},
+
+									"application_security_group_ids": {
+										Type:     schema.TypeSet,
+										Computed: true,
+										Elem:     &schema.Schema{Type: schema.TypeString},
+										Set:      schema.HashString,
+									},
+
+									"primary": {
+										Type:     schema.TypeBool,
+										Computed: true,
+									},
+								},
+							},
+						},
+
+						"dns_servers": {
+							Type:     schema.TypeSet,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+						},
+
+						"internal_dns_name_label": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"applied_dns_servers": {
+							Type:     schema.TypeSet,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+							Set:      schema.HashString,
+						},
+
+						"enable_accelerated_networking": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+
+						"enable_ip_forwarding": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+
+						"private_ip_address": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"private_ip_addresses": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceArmVirtualMachineScaleSetNetworkInterfacesRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.InterfacesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resGroup := d.Get("resource_group_name").(string)
+	vmssName := d.Get("virtual_machine_scale_set_name").(string)
+
+	resp, err := client.ListVirtualMachineScaleSetNetworkInterfaces(ctx, resGroup, vmssName)
+	if err != nil {
+		return fmt.Errorf("Error making Read request on Azure VMSS %q (Resource Group %q): %+v", vmssName, resGroup, err)
+	}
+
+	results := make([]interface{}, 0)
+
+	for _, iface := range resp.Values() {
+		result := make(map[string]interface{})
+
+		if iface.Name != nil {
+			result["name"] = *iface.Name
+		} else {
+			result["name"] = ""
+		}
```

Comment:
Usually we would write this like this:
```suggestion
                name := "" // var name string also works
		if iface.Name != nil {
			name = *iface.Name
		}
		
		// other fields
		
		results = append(results, map[string]interface{}{
		      "name": name,
		      // other fields
		})
```